resources/services/apigateway/rest_apis.go

```diff
@@ -914,18 +916,31 @@ func resolveApigatewayRestAPIModelModelTemplate(ctx context.Context, meta schema
 	if !ok {
 		return fmt.Errorf("expected RestApi but got %T", r)
 	}
-	client := meta.(*client.Client)
-	svc := client.Services().Apigateway
+	cl := meta.(*client.Client)
+	svc := cl.Services().Apigateway
+
+	if api.Id == nil || r.Name == nil {
+		return nil
+	}
 
 	config := apigateway.GetModelTemplateInput{
 		RestApiId: api.Id,
 		ModelName: r.Name,
 	}
 
 	response, err := svc.GetModelTemplate(ctx, &config, func(options *apigateway.Options) {
-		options.Region = client.Region
+		options.Region = cl.Region
 	})
 	if err != nil {
+		if cl.IsNotFoundError(err) {
+			return nil
+		}
+
+		var ae smithy.APIError
+		errorCode := ae.ErrorCode()
```

Comment:
Good that we have integration test :) This code panics as it just accessing nil. 

https://github.com/cloudquery/cq-provider-aws/runs/5598183118?check_suite_focus=true#step:8:2133